### ISSUE TITLE
Closing and reopening DB Editor should reload the database

### ISSUE THREAD

**Issue by DillonJ** [spine-tools/Spine-Toolbox#2307]
If I have DB editor open and a tool runs that writes to the data, I don't see the new data (this is not the issue here). If I close DB Editor and open it again, I still do not see the new data. However, if I reload the datastore, only then is the data truly refreshed.

I believe that closing and reopening a datastore should have the same effect as reloading the datastore

This is happening in version 0.7.

**Comment by manuelma** [comment]
So the detailed step to see this are as follows:

1. Open a DB that has a few thousands parameter values (so that the table is fetched in batches, we use batches of 1000 items at the moment).
2. No need to run a tool or anything like that.
3. Without scrolling the parameter value tables at all, expand an object class in the object tree *without selecting it*. The idea is not to trigger filtering in the tables yet.
4. Select one unit in the tree - now, triggering the filtering. Hopefully, the unit has parameter values that are located near the end of the table.
5. Eventually, the parameter values do not show in the relationship parameter value table (and sometimes neither in the object parameter value table).

**Comment by DillonJ** [comment]
Thanks @manuelma 

@jkiviluo @soininen @PekkaSavolainen This was a major annoyance and it's great to have it fixed. I wonder should we consider merging into v0.7 so others can benefit? I think it might be a while before people can switch to v0.8 so I think where we see cheap oppotunities to improve 0.7 it's worth it?

**Comment by soininen** [comment]
I agree there are bugs in 0.7 that would be very nice to get fixed. How to do that without disturbing the 0.8-dev branch, though? My initial thought was that we could have a 0.7 branch where we applied fixes. This branch would be a "dead end" in the sense that we never merged it back to master. We would just let it wither after we merged 0.8-dev into master. The problem is, how to get users switch branches from master to 0.7 temporarily? Almost everyone I know has cloned the Git repository and gets their updates from master.

(As a side note, graph_view_grid branch looks suspiciously like the 0.7 branch, does it not? ;)

Another idea is to keep developing 0.7 in master but keep the scope small, i.e. mostly bug fixes only. Keep that a "dead end" as well: never merge master to 0.8-dev and once we are confident enough to merge 0.8-dev back to master, resolve all conflicts in favor of 0.8-dev.

I am sure there are other options that I have missed, too.

**Comment by manuelma** [comment]
I was thinking that if we advance master independently of 0.8, then the two will start diverging of course. But then right before merging 0.8 into master we can just revert all the commits in master after the divergence - so 0.8 will be merged without conflicts. Does that make sense?

Or, instead of reverting we can hard reset master to the last commit before the divergence (last common ancestor between 0.8 and master).

**Comment by soininen** [comment]
> Or, instead of reverting we can hard reset master to the last commit before the divergence (last common ancestor between 0.8 and master).

We should *never ever* hard reset a public branch like master.

Reverting the commits back to the last commit that was merged into 0.8-dev sounds like workable solution. We could perhaps tag the current HEAD of master so we know where to start, or?

**Comment by jkiviluo** [comment]
We should fix very annoying bugs in v0.7. Not sure how to judge what is very annoying though - seems like this one is.

**Comment by soininen** [comment]
I have now tagged the last commits in `master` branches that have been merged into `0.8-dev` in all four repositories by `last-merged-to-0.8-dev`.
